### PR TITLE
CT 116: Sort Search Results to prioritize preferrred providers. #19278

### PR DESCRIPTION
**Pull request by zurbergram**
## Description 

https://github.com/department-of-veterans-affairs/vets.gov-team/issues/19278

As a Comparison Tool user, I need "preferred providers" to be sorted to the top of the search results page so that I can see the prioritized providers while deciding on which program / provider to attend.

## Testing
- local testing

## Acceptance Criteria
- [ ] VET TEC search results are sorted according to the following criteria:
     - "Preferred providers" are displayed before non preferred providers first.
     - All cards are sorted alphabetically by provider name third.

### CODE REVIEW

**Review by hughes-dustin — APPROVED** [review]
LGTM

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Review by delli-gatti-michael — APPROVED** [review]
Passes QA

**Comment by zurbergram** [comment]
@department-of-veterans-affairs/backend-review-group Could I get a code review on this at your earliest convenience. Thank you!

**Comment by jholton** [comment]
Can we get a spec in `spec/request/institutions_request_spec`?

**Comment by zurbergram** [comment]
> spec/request/institutions_request_spec

added test

File: spec/request/institutions_request_spec.rb [hunk 1]
```diff
@@ -34,6 +34,38 @@
...
+                             version: Version.current_production.number,
+                             institution: 'D')
+      get(v0_institutions_path(vet_tec_provider: true))
+
+      data_institution_c = JSON.parse(response.body)['data'][0]
+      data_institution_d = JSON.parse(response.body)['data'][1]
+      data_institution_a = JSON.parse(response.body)['data'][2]
+      data_institution_b = JSON.parse(response.body)['data'][3]
+
+      puts data_institution_a
```
Comment (jholton) [inline spec/request/institutions_request_spec.rb]: Remove the `puts` please.

**Review by jholton — APPROVED** [review]
LGTM

**Pull request merged**